app/Libraries/Markdown/StyleBlock/Element.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries\Markdown\StyleBlock;
+
+use League\CommonMark\Block\Element\AbstractBlock;
+use League\CommonMark\Cursor;
+
+class Element extends AbstractBlock
+{
+    /**
+     * @var string
+     */
+    protected $class;
+
+    /**
+     * @var static[]
+     */
+    protected $containedStyleBlocks;
+
+    public function __construct(string $class)
+    {
+        $this->class = $class;
+        $this->containedStyleBlocks = [];
```

Comment:
do it on the declaration above?

Reply:
seen this style somewhere else, woops. setting in declaration is always preferred if possible?

Reply:
I think so unless the value can't be set there (function call etc).